cmd/node-termination-handler.go

```diff
@@ -267,6 +268,11 @@ func drainOrCordonIfNecessary(interruptionEventStore *interruptioneventstore.Sto
 		if nthConfig.CordonOnly || drainEvent.IsRebalanceRecommendation() {
 			err := node.Cordon(nodeName)
 			if err != nil {
+				if errors.IsNotFound(err) {
```

Comment:
@haugenj found another situation where events for other nodes that do not belong to this cluster could bubble up an error - I think for this case we can only do best effort to ignore such cases without removing all the safety guards

Reply:
I think this is the right approach. Originally I was thinking we would totally remove the `os.Exit(1)` here, so your solution feels like a better middle-ground